nightfall-optimist/src/event-handlers/transaction-submitted.mjs

```diff
@@ -71,7 +71,7 @@ async function transactionSubmittedEventHandler(eventParams) {
       saveTransaction({ ...transaction });
     }
 
-    await checkTransaction(transaction, true);
+    await checkTransaction(transaction, { isAlreadyInL2: true, isAlreadyInMempool: true });
```

Comment:
Transactions are now checked at reception and during block assembly. I understand why it’s necessary to do as late as possible. But then, can we just remove the check when transactions are received?

Reply:
@druiz0992 
I think intention is to keep check at the time receive is because, consider a case where proposer is a bad proposer who create a bad-block(overriding the code of check of block assembly - for example) we still need transaction check at receive event-handler in optimist. also if you notice once check fails only we trigger create challenge (is challenger exist in optimist)